### PR TITLE
added angular acceleration computation function and augmented msg con…

### PR DESCRIPTION
**Pull request by mikeallenspach**
I have added a function to compute the angular acceleration from the rotation vector to include it in the commanded trajectory.
Also, I added the angular acceleration to the conversion msgMultiDofJointTrajectoryPointFromEigen.

### CODE REVIEW

File: mav_msgs/include/mav_msgs/common.h [hunk 1]
```diff
@@ -158,11 +158,50 @@ inline Eigen::Vector3d omegaFromRotationVector(
...
+  double phi        = rot_vec.norm();
+  double phi_dot    = rot_vec.dot(rot_vec_vel)/phi; 
+
+  if (abs(phi) < 1.0e-3) {
+    // This captures the case of zero rotation
+    return rot_vec_acc;
+  }
+  // Create skew-symmetric matrix from rotation vector and velocity
+  Eigen::Matrix3d phi_matrix;
+  phi_matrix        << 0.0f, -rot_vec(2), rot_vec(1),
```
Comment (rikba) [inline mav_msgs/include/mav_msgs/common.h]: Please define a "skewSymmetricMatrix" function so you don't have to hardcode it every time.

File: mav_msgs/include/mav_msgs/common.h [hunk 2]
```diff
@@ -158,11 +158,50 @@ inline Eigen::Vector3d omegaFromRotationVector(
...
+  phi_dot_matrix    << 0.0f, -rot_vec_vel(2), rot_vec_vel(1),
+                    rot_vec_vel(2), 0.0f, -rot_vec_vel(0),
+                    -rot_vec_vel(1), rot_vec_vel(0), 0.0f;
+
+
+  // Set up matrices to calculate omega
+  Eigen::Matrix3d W_vel;
+  Eigen::Matrix3d W_acc;
+  W_vel =   phi_matrix * (phi*sin(phi) - 2.0f + 2.0f*cos(phi)) * phi_dot/phi/phi/phi
+            + phi_matrix * phi_matrix * (-2.0f*phi - phi*cos(phi) + 3.0f*sin(phi)) * phi_dot/phi/phi/phi/phi
```
Comment (rikba) [inline mav_msgs/include/mav_msgs/common.h]: 80 character per line. Consider using clang google style formater.

File: mav_msgs/include/mav_msgs/common.h [hunk 3]
```diff
@@ -158,11 +158,50 @@ inline Eigen::Vector3d omegaFromRotationVector(
...
+  Eigen::Matrix3d phi_dot_matrix;
+  phi_dot_matrix    << 0.0f, -rot_vec_vel(2), rot_vec_vel(1),
+                    rot_vec_vel(2), 0.0f, -rot_vec_vel(0),
+                    -rot_vec_vel(1), rot_vec_vel(0), 0.0f;
+
+
+  // Set up matrices to calculate omega
+  Eigen::Matrix3d W_vel;
+  Eigen::Matrix3d W_acc;
+  W_vel =   phi_matrix * (phi*sin(phi) - 2.0f + 2.0f*cos(phi)) * phi_dot/phi/phi/phi
```
Comment (rikba) [inline mav_msgs/include/mav_msgs/common.h]: Maybe that's nitpicking but you may want to define 1/phi, 1/phi/phi, ... as constants such that you use less divisions, i.e.,
```
const double kPhiInv = 1.0 / phi;
const double kPhi2Inv = kPhiInv / phi;
const double kPhi3Inv = kPhi2Inv / phi;
const double kPhi4Inv = kPhi3Inv / phi;
```

File: mav_msgs/include/mav_msgs/common.h [hunk 4]
```diff
@@ -158,11 +158,50 @@ inline Eigen::Vector3d omegaFromRotationVector(
                 -rot_vec(1), rot_vec(0), 0.0f;
   // Set up matrix to calculate omega
   Eigen::Matrix3d W;
-  W = Eigen::MatrixXd::Identity(3,3) - phi_matrix * (1-cos(phi))/phi/phi
+  W = Eigen::MatrixXd::Identity(3,3) + phi_matrix * (1-cos(phi))/phi/phi
```
Comment (rikba) [inline mav_msgs/include/mav_msgs/common.h]: Maybe add the reference also here, so people know where this formula comes from.

File: mav_msgs/include/mav_msgs/common.h [hunk 5]
```diff
@@ -58,6 +58,7 @@ inline Eigen::Vector3d vector3FromPointMsg(const geometry_msgs::Point& msg) {
   return Eigen::Vector3d(msg.x, msg.y, msg.z);
 }
 
+
```
Comment (rikba) [inline mav_msgs/include/mav_msgs/common.h]: Remove unnecessary line.

File: mav_msgs/include/mav_msgs/common.h [hunk 6]
```diff
@@ -142,27 +143,91 @@ inline void getEulerAnglesFromQuaternion(const Eigen::Quaternion<double>& q,
   }
 }
 
+inline void skewMatrixFromVector(   const Eigen::Vector3d& vec, 
+                                    Eigen::Matrix3d* vec_skew){
+    assert( vec_skew );
```
Comment (rikba) [inline mav_msgs/include/mav_msgs/common.h]: Remove whitespace.

File: mav_msgs/include/mav_msgs/common.h [hunk 7]
```diff
@@ -142,27 +143,91 @@ inline void getEulerAnglesFromQuaternion(const Eigen::Quaternion<double>& q,
   }
 }
 
+inline void skewMatrixFromVector(   const Eigen::Vector3d& vec, 
+                                    Eigen::Matrix3d* vec_skew){
```
Comment (rikba) [inline mav_msgs/include/mav_msgs/common.h]: Format whitespace.

File: mav_msgs/include/mav_msgs/common.h [hunk 8]
```diff
@@ -142,27 +143,91 @@ inline void getEulerAnglesFromQuaternion(const Eigen::Quaternion<double>& q,
...
+    assert( vec_skew );
+    Eigen::Matrix3d vec_skew_;
```
Comment (rikba) [inline mav_msgs/include/mav_msgs/common.h]: Avoid copy, i.e.,
```
*vec_skew << 0.0f, -vec(2), vec(1),
                    vec(2), 0.0f, -vec(0),
                    -vec(1), vec(0), 0.0f;
```

File: mav_msgs/include/mav_msgs/common.h [hunk 9]
```diff
@@ -142,27 +142,85 @@ inline void getEulerAnglesFromQuaternion(const Eigen::Quaternion<double>& q,
...
 }
 
+inline void skewMatrixFromVector(const Eigen::Vector3d& vec,
+                                 Eigen::Matrix3d* vec_skew) {
+  assert(vec_skew);
+  Eigen::Matrix3d vec_skew_;
+  vec_skew_ << 0.0f, -vec(2), vec(1), vec(2), 0.0f, -vec(0), -vec(1), vec(0),
+      0.0f;
+
+  *vec_skew = vec_skew_;
```
Comment (rikba) [inline mav_msgs/include/mav_msgs/common.h]: Can you not avoid the copy and do
```
*vec_skew << << 0.0f, -vec(2), vec(1), vec(2), 0.0f, -vec(0), -vec(1), vec(0),
      0.0f;
```
?

File: mav_msgs/include/mav_msgs/common.h [hunk 10]
```diff
@@ -142,27 +142,85 @@ inline void getEulerAnglesFromQuaternion(const Eigen::Quaternion<double>& q,
...
+    const Eigen::Vector3d& rot_vec_acc) {
+  double phi = rot_vec.norm();
+  double phi_dot = rot_vec.dot(rot_vec_vel) / phi;
+
+  double phi_inv = 1.0 / phi;
+  double phi_2_inv = phi_inv / phi;
+  double phi_3_inv = phi_2_inv / phi;
+  double phi_4_inv = phi_3_inv / phi;
+
+  if (abs(phi) < 1.0e-3) {
```
Comment (rikba) [inline mav_msgs/include/mav_msgs/common.h]: You should do this test right after calculating `phi`.

**Comment by mikeallenspach** [comment]
Great, thank you for the review and helpful inputs.

**Pull request merged**